rust-mode.el

```diff
@@ -1195,12 +1195,24 @@ This is written mainly to be used as `end-of-defun-function' for Rust."
   (with-current-buffer (get-buffer-create "*rustfmt*")
     (erase-buffer)
     (insert-buffer-substring buf)
-    (if (zerop (call-process-region (point-min) (point-max) rust-rustfmt-bin t t nil))
-        (progn
-          (if (not (string= (buffer-string) (with-current-buffer buf (buffer-string))))
+    (let ((tmpf (make-temp-file "rustfmt")))
+      (let ((ret (call-process-region (point-min) (point-max) rust-rustfmt-bin
+                                      t `(t ,tmpf) nil)))
+        (cond
+         ((zerop ret)
+          (error "Rustfmt failed, see *rustfmt* buffer for details"))
+         ((= ret 3)
```

Comment:
Yes, it's documented in the last paragraph here in the README: https://github.com/rust-lang-nursery/rustfmt/blob/master/README.md#running